cosmos/ingestion/ingest/process/aggregation/reaggregate_equations.py

```diff
@@ -6,21 +6,75 @@
 from ..proposals.connected_components import get_proposals
 from PIL import Image
+import fitz
+import re
 
+COSMOS_DOC_HEIGHT=1920
+LABEL_PATTERN=re.compile(r'[\(ð][1-9A-Z]+.?[0-9A-Za-z]*[\)Þ]',re.MULTILINE)
+HI_RES_DPI=300
 
-RIGHT_JUSTIFY_EDGE = 15 # Farthest from an image's right edge that a label can be
-LABEL_MAX_WIDTH = 60 # Max width of a label
-LABEL_MAX_HEIGHT = 40 # Max height of a label
+LABEL_OVERLAP = 0.9 # Remove any segments that overlap at least this % with a label bb
 PADDING = 5 # Pad the very narrowly cropped equation segments back out a bit
 
 # Convenience class to assign property names to the 4 tuple of ints used for segment coordinates
 Bounds = namedtuple("Bounds", ("left", "top", "right", "bottom"))
 
-def middle(bounds: Bounds):
+def _get_label_bbs(pymu_page: fitz.Page, labels: List[str], reference_frame: Bounds) -> List[Bounds]:
+    """ Given a list of label texts on a page, find the bounding boxes
+    for those label texts. Then, convert from PyMuPDF coordinates back to COSMOS coordinates
+    """
+    # TODO clean up frame of reference shifting
+    x0, y0, *_ = reference_frame
+    height_ratio = (pymu_page.rect[3] - pymu_page.rect[1]) / COSMOS_DOC_HEIGHT
+    label_bbs = []
+    for label in labels:
+        label_bbs.extend(pymu_page.search_for(label))
+    unscaled_bbs = [b/height_ratio for b in label_bbs]
+    return [Bounds(b.x0 - x0, b.y0 - y0, b.x1 - x0, b.y1 - y0) for b in unscaled_bbs]
+
+def _get_scaled_rect(pymu_page: fitz.Page, bounds: Bounds) -> Tuple[fitz.Page, fitz.Rect]:
+    """ Given a bounding box in the COSMOS coordinate system and a PyMuPDF page,
+    return that bounding box in the PyMuPDF coordinate system """
+    height_ratio = (pymu_page.rect[3] - pymu_page.rect[1]) / COSMOS_DOC_HEIGHT
+    return fitz.Rect(*bounds) * height_ratio
+
+# TODO don't manually reimplement basic geometry functions
+
+def _middle(bounds: Bounds):
     """Get the vertical center of a Bounds"""
     return (bounds.top + bounds.bottom) / 2
 
-def pad(bound: Bounds, padding: int = PADDING):
+def _overlap(r1: Bounds, r2: Bounds):
+    """ Get the overlapping area between two Bounds"""
+    return max(0, min(r1.right, r2.right) - max(r1.left, r2.left)) * \
+        max(0, min(r1.bottom, r2.bottom) - max(r1.top, r2.top))
+
+def _area(r1: Bounds):
+    """ Get the area of the Bounds """
+    return (r1.right - r1.left) * (r1.bottom - r1.top)
+
+def _overlaps_with_label(bounds: Bounds, label_bounds: List[Bounds]):
+    """ Return whether a Bounds is largely overlapping with a label """
+    area = _area(bounds)
+    return any(_overlap(bounds, lb) > 0.9 * area for lb in label_bounds)
+
+def find_labels_for_equation(pymu_page: fitz.Page, bounds: Bounds):
+    """ Find the text in the text layer matching `(X.Y)` that is in close proximity 
+    to the equation bounding box """
+    rect = _get_scaled_rect(pymu_page, bounds)
+    # guess which "half" of the page the equation is on, then look for a label
```

Comment:
Hmm. Did we test this on any of the preprints? They're typically single-column, so I think they might cause headaches here. For example:

![image](https://github.com/UW-COSMOS/Cosmos/assets/1444733/ae22e82e-61ba-4bbd-b3e2-dc4ca8289825)

(from 5ebbf1c0998e17af826e5068)

Reply:
Just confirmed that the correct label is found in these cases, however it frequently omits the "Arithmetic mean = " half